test/e2e/framework/framework.go

```diff
@@ -184,14 +183,18 @@ func NewFramework(baseName string, options Options, client clientset.Interface)
 	})
 
 	ginkgo.BeforeEach(f.BeforeEach)
-	ginkgo.AfterEach(f.AfterEach)
 
 	return f
 }
 
 // BeforeEach gets a client and makes a namespace.
 func (f *Framework) BeforeEach() {
-	f.beforeEachStarted = true
+	// DeferCleanup, in constrast to AfterEach, triggers execution in
+	// first-in-last-out order. This ensures that the framework instance
+	// remains valid as long as possible.
+	//
+	// In addition, AfterEach will not be called if a test never gets here.
+	ginkgo.DeferCleanup(f.AfterEach)
```

Comment:
will this run twice? we add it to `	f.cleanupHandle = AddCleanupAction(f.AfterEach)` in the next line

Reply:
it sounds like the thing to remove is the `cleanupHandle` , seems that `DeferCleanup` just fixes https://github.com/onsi/ginkgo/v2/issues/222

Reply:
I think it removes itself once it ran once. But you are right, let's use this opportunity to also remove cleanupHandle.

Reply:
Done.